### DISCUSSION TITLE
Prompt Library

### DISCUSSION THREAD

**Discussion by emshvac** [cline/cline#401] [Feature Requests]
I find myself using certain prompts for example explaining what it needs to look at and include as well as expected output for certain functions.  It would be nice to have a library of prompts that you frequently use as a shortcut.

**Comment by tsdevau** [comment]
This would be a very useful addition!

**Comment by AlexJ-StL** [comment]
This would be a fantastic use of macros/presets. You could have a fixed number of user set presets that could be set to user defined macros in VS Code or whatever IDE people are using Claude Dev...or Cline now I suppose... I like the idea though.

I don't really have much use for this at the IDE level, but I have been building my prompt library for about 2 years now.... Let me tell you... It is such a time saver and eye opening to how repetitive the tasks that I used to do are... I thought the library would get unruly so I created a local semantic vector database and prepared myself for painful database management... Yeah.... That is the emptiest 2 year old database I've personally ever seen. I could have just as easily stored them in a spreadsheet.... Scratch that... A spreadsheet would be easier and now that I type this I think I am going to start using a spreadsheet instead.

**Reply by wilkax** [reply]
You mind sharing some? I just started with AI-based programming and often wonder what the best way to ask a certain thing is (like „reduce the unused whitespace between x and y“)

**Reply by AlexJ-StL** [reply]
I would be more than willing to help get you started, but I will do better than simply give you a few tricks. After all "Feed a man a fish, you feed him for a day. Teach a man to fish, feed him for a lifetime", right?

Out of respect for the thread let's take this convo elsewhere. I have a few ways to contact me in my profile.... I think... By the time you read this I will have checked to make sure I have some sort of contact info on my profile. Hit me up and I will give you some resources to get you on your way.

**Reply by wilkax** [reply]
...tried on Instagram, unfortunately without success so far - please have a look in the message folders of which you're not automatically getting notified ;-)

**Reply by AlexJ-StL** [reply]
I am very sorry for not checking my Instagram. December turned into craziness out of nowhere. I will check it now and get back with you.

**Comment by philikai** [comment]
+1 on this. It would be great to have different prompt mechanisms as well as pointing it to a project requirement markdown file by default. 

I often stay on a thread with Cline longer, as I do not want to waste new API calls and time to describe my project again. 

However, this often leads to impaired code output, when I am defaulting to: "Describe what you did so far and what needs to be done for the next developer working on this project tomorrow" :D

**Reply by AlexJ-StL** [reply]
I'm just thinking out loud here, but in the event that it would help you... Couldn't you adopt Aider's mechanism? Aider uses a project tree and Git comments when going back into an existing code base. I have been using Cline/Claude Dev to start new projects and Aider to edit them personally and find that work flow to be most fluid. Cline's strengths seem to me to be creative or higher level tasks. Aider is absolutely amazing at refactoring, feature addition, and general editing of specifically identified code sections. With that said, I wonder if your issue might get worked out by feeding Cline a project tree and the git change log at the beginning of any section which could be held in its cached context. I suppose you could also use some kind of mechanism that references a local memory database (likely RAG) as well.

**Comment by calberts** [comment]
I have a lot of markdown prompts in prompt directory,  for example, for functions, but also for the main applications description, and it's hard to keep them up to date. Today ask  ai to generate a propmpt from the code when the code is ready this should be automated, So I like that there is an easy way to maintain and update the prompts after the conversation task ik cline. the aim should be when executing all the prompts the application will be automatically generated.   So yes this would be a very useful addition!

**Reply by slavakurilyak** [reply]
Add the following prompt to your `.clinerules` to direct cline to update your prompts after code execution:

```
# .clinerules
After each code change:

1. Analyze modified code files to extract core functionality and patterns
2. Update corresponding prompts in prompts/ directory or prompts.md to reflect current implementation

Skip prompt updates for minor changes (formatting, comments, variable renames).
```

This is a suggestion only

I encourage you to report your results below

**Comment by profplum700** [comment]
For us real duffers out there who are lost for what to do next, I think it would be great for Cline to also, when it sees good reason to, to suggest one or more of these prompts to run next. It already often suggests next steps. Get it to suggest next prompts!
